### ISSUE TITLE
Scraping for links - doc.All/QuerySelectorAll("*") returns only three nodes

### ISSUE THREAD

**Issue by harvinders** [AngleSharp/AngleSharp#767]
### Prerequisites

- [x] Can you reproduce the problem in a [anglesharp.github.io](https://anglesharp.github.io/projects.html)?
- [X] Are you running the latest version of AngleSharp?
- [x] Did you check the FAQs to see if that helps you?
- [X] Are you reporting to the correct repository? (there are multiple AngleSharp libraries, e.g., `AngleSharp.Css` for CSS support)
- [X] Did you perform a search in the issues?

For more information, see the `CONTRIBUTING` guide.

### Description
I have written following to look for all the links in the document, but the `where` is hit only thrice for html, head and body elements. My understanding is based on the [Examples](https://github.com/AngleSharp/AngleSharp/wiki/Examples#getting-certain-elements)

            HtmlParser parser = new HtmlParser();
            var doc = await parser.ParseDocumentAsync("https://anglesharp.github.io/projects.html");
            foreach (var link in doc.QuerySelectorAll("*")
                .Where(x => x.LocalName == "a")
                .Select(x => x.GetAttribute("href"))
                .Where(x => null != x))
            {
                _testOutputHelper.WriteLine(link);
            }

**Expected behavior:** Should get all the links

**Actual behavior:** Getting only three

**Environment details:** Windows 10, 1803, Visual Studio 2019

**Comment by FlorianRappl** [comment]
You have no requester set up. Can you please follow the examples more closely and try again?

TL;DR: Use a `BrowsingContext` and configure it properly (e.g., with `Configuration.Default.WithDefaultLoader()`).
